sandbox/apparmor/notify/listener/listener.go

```diff
@@ -323,18 +298,20 @@ func (l *Listener) Run() error {
 	// is called) return and close the tomb's dead channel, as it is the last
 	// and only tracked goroutine.
 	l.tomb.Go(func() error {
-		var err error
 		for {
-			err = l.tomb.Err()
-			if err != tomb.ErrStillAlive {
+			if l.tomb.Err() != tomb.ErrStillAlive {
 				break
```

Comment:
shouldn't we also log this error or is it logged at another level?

Reply:
Since we've replaced the error returns from tracked goroutines with logs instead, the only error should be the listener being closed. But I don't see that logged elsewhere, so you're right, would be good to log here.